contracts/src/base/registry/facets/distribution/v2/StakingRewards.sol

```diff
@@ -0,0 +1,490 @@
+    address proxy = ds.proxyById[depositId];
+    ds.stakeToken.safeTransferFrom(
+      proxy,
+      deposit.owner,
+      deposit.pendingWithdrawal
+    );
+  }
+
+  function claimReward(
+    Layout storage ds,
+    address beneficiary,
+    address recipient
+  ) internal returns (uint256 reward) {
+    updateGlobalReward(ds);
+
+    Treasure storage treasure = ds.treasureByBeneficiary[beneficiary];
+    updateReward(ds, treasure);
+
+    reward = treasure.unclaimedRewardSnapshot;
+    if (reward != 0) {
+      treasure.unclaimedRewardSnapshot = 0;
+      ds.rewardToken.safeTransfer(recipient, reward);
+    }
+  }
+
+  function notifyRewardAmount(Layout storage ds, uint256 reward) internal {
+    ds.rewardPerTokenAccumulated = currentRewardPerTokenAccumulated(ds);
+
+    // cache storage reads
+    (uint256 rewardDuration, uint256 rewardEndTime) = (
+      ds.rewardDuration,
+      ds.rewardEndTime
+    );
+
+    uint256 rewardRate = FixedPointMathLib.fullMulDiv(
+      reward,
+      SCALE_FACTOR,
+      rewardDuration
+    );
+    // if the reward period hasn't ended, add the remaining reward to the reward rate
+    if (rewardEndTime > block.timestamp) {
+      uint256 remainingTime;
+      unchecked {
+        remainingTime = rewardEndTime - block.timestamp;
+      }
+      rewardRate += FixedPointMathLib.fullMulDiv(
+        ds.rewardRate,
+        remainingTime,
+        rewardDuration
+      );
+    }
+
+    // batch storage writes
+    (ds.rewardEndTime, ds.lastUpdateTime, ds.rewardRate) = (
+      block.timestamp + rewardDuration,
+      block.timestamp,
+      rewardRate
+    );
+
+    if (rewardRate < SCALE_FACTOR) {
+      CustomRevert.revertWith(StakingRewards__InvalidRewardRate.selector);
+    }
+
+    if (
+      FixedPointMathLib.fullMulDiv(rewardRate, rewardDuration, SCALE_FACTOR) >
+      IERC20(ds.rewardToken).balanceOf(address(this))
+    ) {
+      CustomRevert.revertWith(StakingRewards__InsufficientReward.selector);
+    }
+  }
+
+  function _increaseEarningPower(
+    Layout storage ds,
+    Deposit storage deposit,
+    Treasure storage beneficiaryTreasure,
+    uint96 amount,
+    address delegatee,
+    uint256 commissionRate
+  ) private {
+    unchecked {
+      if (commissionRate == 0) {
+        beneficiaryTreasure.earningPower += amount;
+      } else {
+        uint96 commissionEarningPower = uint96(
+          (uint256(amount) * commissionRate) / MAX_COMMISSION_RATE
+        );
+        deposit.commissionEarningPower += commissionEarningPower;
+        beneficiaryTreasure.earningPower += amount - commissionEarningPower;
+
+        Treasure storage delegateeTreasure = ds.treasureByBeneficiary[
+          delegatee
+        ];
+        updateReward(ds, delegateeTreasure);
+        delegateeTreasure.earningPower += commissionEarningPower;
+      }
+    }
+  }
+
+  function _decreaseEarningPower(
+    Layout storage ds,
+    Deposit storage deposit,
+    Treasure storage beneficiaryTreasure
+  ) private {
```

Comment:
internal?

Reply:
They are intended to be private and not used outside or individually.